### ISSUE TITLE
zlib version

### ISSUE THREAD

**Issue by dimpase** [ibara/baseutils#3]
Might be a bit offtopic here, but why does OpenBSD not upgrade to a newer zlib. 1.2.3 is really ancient (from 2005), and it causes e.g. a problem with R - which explcitly requires at least 1.2.5, see e.g.
https://github.com/wch/r-source/blob/0fac8b9c4d06fafcbd3f74e35fa30107286fcd7a/m4/R.m4#L3010

Yes, you can patch the R test away, but that's really a wrong way IMHO.

**Comment by ibara** [comment]
Seems like a question better suited for the mailing lists. I certainly don't have an answer.

**Comment by dimpase** [comment]
I found http://openbsd-archive.7691.n7.nabble.com/PATCH-update-zlib-to-1-2-8-td262546.html#a262548 (back from 2014) which contains a toxic rant on uselessness of newer than 1.2.3 releaseas of zlib, and this is despite 1.2.4 and 1.2.5 fixing memory leaks, bugs, etc.

I really don't know what to make of it - is this deemed a highly non-essential update?

**Comment by ibara** [comment]
I would say to be careful when reading the mailing lists. Anyone can say anything. You'll note that the only responses by actual OpenBSD developers were:
1. sthen@ mentioning his experience attempting an update to zlib, which ultimately did not happen because he did not update the kernel copy of zlib.
2. espie@ talking about the surprising difficulty of updating zlib.

Anything else in that thread that doesn't have a diff attached to it is irrelevant. Words can be cheap and are unfortunately the cheap words are sometimes in great supply on the lists by non-devs.

That being said, even though I am an OpenBSD developer, I can't answer the question of whether or not an update to zlib is essential or not since it's not my area of expertise nor do I have all the architectures needed to do a full and proper testing. What I can do is give you my thought process for my portable versions of OpenBSD code. My philosophy is that users should get the OpenBSD experience--good and bad--since that is what those users are expecting. It's the same reason that I direct all PRs not directly addressing the addition or maintenance of other platforms to the tech@ mailing list. The userland version of zlib found in OpenBSD is certainly sufficient for OpenBSD's tools. If through the use of this software, improvements to OpenBSD's version of zlib can be found, then that's great and I'd encourage those improvements to be sent to tech@ so we can all benefit.

**Comment by dimpase** [comment]
Thanks -  it's hard to see which posts are from outsiders, and which are not.

As I user, I expect non-buggy versions of libraries to be available. It also puzzles me that a potential contributor is apparently expected to do all the testing (which might be not possible due to lack of right hardware).

**Comment by ibara** [comment]
I don't think it was suggested that anyone single-handedly perform all the testing. What was asked is that the potential contributor coordinate testing since the last thing we would want to do is spend all that time on an update only to find it breaks on luna88k because no one asked for testing there. I think that's a reasonable approach.

**Comment by ghost** [comment]
I've made work on a personal fork that unvendors zlib. However, it's only used in grep: I removed compress and the like.

**Comment by ibara** [comment]
OK? That is not congruous with the goals of this repo but you are free to do whatever you'd like.

**Comment by ghost** [comment]
I was hoping to catch the attention of the person who opened the issue, who perhaps wanted  to use this with a newer zlib :pensive:. Seems I may've misjudged intent, however.

**Comment by dimpase** [comment]
well, the thing is that, IIRC, R project merely tests the zlib library version.

**Comment by ibara** [comment]
Also, importantly, the libz here never gets installed on your system. It gets statically linked into the utilities that need it.
